dbms/src/AggregateFunctions/AggregateFunctionQuantile.h

```diff
@@ -49,12 +56,14 @@ template <
     bool returns_many
 >
 class AggregateFunctionQuantile final : public IAggregateFunctionDataHelper<Data,
-    AggregateFunctionQuantile<Value, Data, Name, have_second_arg, FloatReturnType, returns_many>>
+    AggregateFunctionQuantile<Value, Data, Name, has_second_arg, FloatReturnType, returns_many>>
 {
 private:
     using ColVecType = std::conditional_t<IsDecimalNumber<Value>, ColumnDecimal<Value>, ColumnVector<Value>>;
 
-    static constexpr bool returns_float = !std::is_same_v<FloatReturnType, void>;
+    static constexpr bool returns_float = !(std::is_same_v<FloatReturnType, void>)&&(
```

Comment:
Don't return float if `FloatReturnType` is unspecified.
Don't return float if `Value` is `Date` or `DateTime`.
Always return float if `Data` is `QuantileTiming`.